src/mod_privacy_rdbms.erl

```diff
@@ -311,18 +314,47 @@ execute_privacy_list_insert(LServer, LUser, Name) ->
 execute_delete_data_by_id(LServer, ID) ->
     mongoose_rdbms:execute_successfully(LServer, delete_data_by_id, [ID]).
 
-replace_data_rows(LServer, ID, Rows) when is_integer(ID)->
-    execute_delete_data_by_id(LServer, ID),
-    [mongoose_rdbms:execute_successfully(LServer, privacy_data_insert, [ID|Args])
-     || Args <- Rows],
+replace_data_rows(LServer, ID, []) when is_integer(ID) ->
+    %% Just remove the data, nothing should be inserted
+    execute_delete_data_by_id(LServer, ID);
+replace_data_rows(LServer, ID, Rows) when is_integer(ID) ->
+    {selected, OldRows} = execute_privacy_data_get_by_id(LServer, ID),
+    New = lists:sort(Rows),
+    Old = lists:sort([tuple_to_list(Row) || Row <- OldRows]),
+    Diff = diff_rows(ID, New, Old, []),
+    [mongoose_rdbms:execute_successfully(LServer, Q, Args) || {Q, Args} <- Diff],
```

Comment:
Maybe rewrite it to protect against erroneous items in the list that are silently ignored by LC?